collection-scripts/gather_performance_profile_controller

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+
+function ppc_nodes(){
+  
+check_node_gather_pods_ready() {
+   line=$(oc get ds perf-node-gather-daemonset -o=custom-columns=DESIRED:.status.desiredNumberScheduled,READY:.status.numberReady --no-headers -n perf-node-gather)
+
+    IFS=$' '
+    read desired ready <<< $line
+    IFS=$'\n'
+
+    if [[ "$desired" != "0" ]] && [[ "$ready" == "$desired" ]]
```

Comment:
This leave no room for error (IE all pods in the daemonset have to come up and be working). 

I worry about how 'strict this is'. While you can control how many nodes this runs on with `perf-node-gather-daemonset` labels; on systems where you 'expect things to not be working properly' this level of 'stringency' may be problematic. 

That said I understand the chicken & egg situation this creates, we should just look at how we can make this a bit more forgiving (if possible).

Reply:
I second @sferich888 here, it's more reasonable to have some wait period before failing.